### ISSUE TITLE
DOC: Improve migration instructions for `maximum_sctype`

### ISSUE THREAD

**Issue by bersbersbers** [numpy/numpy#25074]
### Issue with current documentation:

It seems that `maximum_sctype` is now deprecated, compare https://numpy.org/neps/nep-0052-python-api-cleanup.html#cleaning-up-the-main-namespace:

> All *sctype functions can be deprecated and removed, they (see [gh-17325](https://github.com/numpy/numpy/issues/17325), [gh-12334](https://github.com/numpy/numpy/issues/12334), and other issues for maximum_sctype and related functions).

Unfortunately,
- the migration guidelines on `maximum_sctype` is empty, https://numpy.org/devdocs/numpy_2_0_migration_guide.html#main-namespace
- https://github.com/numpy/numpy/issues/17325 and https://github.com/numpy/numpy/issues/12334 are not helpful, either
- I find no [other issues](https://github.com/numpy/numpy/issues?q=is%3Aissue+%22maximum_sctype%22) on `maximum_sctype` that are helpful
- `maximum_sctype` was removed in https://github.com/numpy/numpy/pull/24154 (note: "undocumented functions") although `maximum_sctype` [*is* documented](https://numpy.org/doc/stable/reference/generated/numpy.maximum_sctype.html)

In summary, when documented functions are deprecated, I'd expect some assistance in migration.

**Comment by bersbersbers** [comment]
Particularly, it would be good to know how to get the maximum supported unsigned integer type other than using `np.maximum_sctype(np.uint8)`.

By the way, I find it surprising that `min_scalar_type` shall remain while `maximum_sctype` shall be removed.

**Comment by rgommers** [comment]
Thanks @bersbersbers, some migration guidance would be good to add here indeed. Cc @mtsokol can you please look into doing this?

Re `np.maximum_sctype(np.uint8)`, this is obscure code for no reason, I'd just write `np.uint64` instead. This is always the correct answer, and that won't change. And if we ever, for example, would add `np.uint128`, (not going to happen, but just as an example) you really want to make an explicit change to your code there if you want to use it, rather than a silent one, because it can break a lot of things to silently switch over.

**Comment by rgommers** [comment]
This issue was addressed by gh-25081, so closing. Thanks @bersbersbers and @mtsokol.
